### PR TITLE
[HttpKernel][VarDumper] Fix dumping with labels

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Should fix https://github.com/symfony/symfony/pull/50347#issuecomment-1558458542 /cc @bobthecow 

Also improves the display in the WDT:

image

image

### CODE REVIEW

File: src/Symfony/Bundle/DebugBundle/Resources/views/Profiler/dump.html.twig [hunk 1]
```diff
@@ -45,7 +48,12 @@
 
     {% for dump in collector.getDumps('html') %}
         <div class="sf-dump sf-reset">
-            <span class="metadata">In
+            <span class="metadata">
+                {% if dump.label is defined and '' != dump.label %}
+                    <strong>{{ dump.label }}</strong> in
+                {% else %}
+                    In
```
Comment (fancyweb) [inline src/Symfony/Bundle/DebugBundle/Resources/views/Profiler/dump.html.twig:55]: When there's no label we display `In FooController.php line 14` now?

File: src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php [hunk 2]
```diff
@@ -125,7 +128,7 @@ public function collect(Request $request, Response $response, \Throwable $except
             }
 
             foreach ($this->data as $dump) {
-                $this->doDump($dumper, $dump['data'], $dump['name'], $dump['file'], $dump['line']);
+                $this->doDump($dumper, $dump['data'], $dump['name'], $dump['file'], $dump['line'], $dump['label'] ?? '');
```
Comment (fancyweb) [inline src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php:131]: Doesn't the label key always exist (same below)?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php:131]: I thought about existing profiles, which we might load from a previous version during migrations.

File: src/Symfony/Bundle/DebugBundle/Resources/views/Profiler/dump.html.twig (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Bundle/DebugBundle/Resources/views/Profiler/dump.html.twig:55]: as was before, no? (see L48 before)

**Review by stof — APPROVED** [review]

**Review by fancyweb — APPROVED** [review]

**Comment by bobthecow** [comment]
Thanks for jumping on this, @nicolas-grekas! 

This fix works, in that it no longer breaks functionality. But it does look like it introduces a change in formatted output that puts empty spans in front of a lot of things:

Screenshot 2023-05-23 at 10 19 29 AM

Either that or this was introduced somewhere else but masked by the BC break in my tests.

**Comment by nicolas-grekas** [comment]
@bobthecow fixed!

**Comment by bobthecow** [comment]
@nicolas-grekas perfect, thanks so much!

**Pull request merged**